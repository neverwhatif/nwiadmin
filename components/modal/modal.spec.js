import React from 'react';
import { shallow } from 'enzyme';

import { ModalComponent } from './index';

const mockLocation = { pathname: 'lorem' };
const mockNewLocation = { pathname: 'ipsum' };

describe('ModalComponent', () => {
    const component = shallow(<ModalComponent isOpen={false}>Lorem</ModalComponent>);
    const componentOpen = shallow(<ModalComponent isOpen>Lorem</ModalComponent>);
    const componentWithTitle = shallow(<ModalComponent isOpen title="Lorem">Lorem</ModalComponent>);
    const componentWithType = shallow(<ModalComponent isOpen type="lorem">Lorem</ModalComponent>);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentOpen).toMatchSnapshot();
        expect(componentWithTitle).toMatchSnapshot();
        expect(componentWithType).toMatchSnapshot();
    });
    it('should be open', () => {
        expect(component.prop('isOpen')).toBe(false);
        expect(componentOpen.prop('isOpen')).toBe(true);
    });
    it('should be closed', () => {
        componentOpen.setState({ isAllowedOpen: false });
        expect(componentOpen.prop('isOpen')).toBe(false);
    });
    it('should render title', () => {
        expect(componentWithTitle.hasClass('contentHasTitle')).toBe(true);
    });
    it('should render type', () => {
        expect(componentWithType.hasClass('contentLorem')).toBe(true);
    });
    it('should have a default onRequestClose', () => {
        expect(ModalComponent.defaultProps.onRequestClose()).toBe(null);
    });
    it('should handle no location', () => {
        // Kept of the initial component definitions to help with resetting state and props
        const componentWithLocation = shallow(<ModalComponent isOpen location={mockLocation}>Lorem</ModalComponent>);

        componentWithLocation.setProps({ location: null });
        expect(componentWithLocation.state('isAllowedOpen')).toBe(true);
    });
    it('should close on location change', () => {
        // Kept of the initial component definitions to help with resetting state and props
        const componentWithLocation = shallow(<ModalComponent isOpen location={mockLocation}>Lorem</ModalComponent>);

        componentWithLocation.setProps({ location: mockNewLocation });
        expect(componentWithLocation.state('isAllowedOpen')).toBe(false);
    });
});
