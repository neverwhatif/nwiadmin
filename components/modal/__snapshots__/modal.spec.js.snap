// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalComponent
    isOpen={false}
    location={null}
    onRequestClose={[Function]}
    title={null}
  >
    Lorem
  </ModalComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        null,
        <div
          className="scroll"
        >
          Lorem
        </div>,
      ],
      "className": "content",
      "closeTimeoutMS": 0,
      "contentLabel": "Modal",
      "isOpen": false,
      "onRequestClose": [Function],
      "overlayClassName": "overlay",
      "parentSelector": [Function],
      "portalClassName": "root",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "scroll",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          null,
          <div
            className="scroll"
          >
            Lorem
          </div>,
        ],
        "className": "content",
        "closeTimeoutMS": 0,
        "contentLabel": "Modal",
        "isOpen": false,
        "onRequestClose": [Function],
        "overlayClassName": "overlay",
        "parentSelector": [Function],
        "portalClassName": "root",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "scroll",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ModalComponent should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalComponent
    isOpen={true}
    location={null}
    onRequestClose={[Function]}
    title={null}
  >
    Lorem
  </ModalComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        null,
        <div
          className="scroll"
        >
          Lorem
        </div>,
      ],
      "className": "content",
      "closeTimeoutMS": 0,
      "contentLabel": "Modal",
      "isOpen": true,
      "onRequestClose": [Function],
      "overlayClassName": "overlay",
      "parentSelector": [Function],
      "portalClassName": "root",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "scroll",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          null,
          <div
            className="scroll"
          >
            Lorem
          </div>,
        ],
        "className": "content",
        "closeTimeoutMS": 0,
        "contentLabel": "Modal",
        "isOpen": true,
        "onRequestClose": [Function],
        "overlayClassName": "overlay",
        "parentSelector": [Function],
        "portalClassName": "root",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "scroll",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ModalComponent should render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalComponent
    isOpen={true}
    location={null}
    onRequestClose={[Function]}
    title="Lorem"
  >
    Lorem
  </ModalComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        <div
          className="title"
        >
          <FormLegend
            isOffset={true}
          >
            Lorem
          </FormLegend>
        </div>,
        <div
          className="scroll scrollHasTitle"
        >
          Lorem
        </div>,
      ],
      "className": "content",
      "closeTimeoutMS": 0,
      "contentLabel": "Modal",
      "isOpen": true,
      "onRequestClose": [Function],
      "overlayClassName": "overlay",
      "parentSelector": [Function],
      "portalClassName": "root",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormLegend
            isOffset={true}
          >
            Lorem
          </FormLegend>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Lorem",
            "isOffset": true,
          },
          "ref": null,
          "rendered": "Lorem",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "scroll scrollHasTitle",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          <div
            className="title"
          >
            <FormLegend
              isOffset={true}
            >
              Lorem
            </FormLegend>
          </div>,
          <div
            className="scroll scrollHasTitle"
          >
            Lorem
          </div>,
        ],
        "className": "content",
        "closeTimeoutMS": 0,
        "contentLabel": "Modal",
        "isOpen": true,
        "onRequestClose": [Function],
        "overlayClassName": "overlay",
        "parentSelector": [Function],
        "portalClassName": "root",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormLegend
              isOffset={true}
            >
              Lorem
            </FormLegend>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Lorem",
              "isOffset": true,
            },
            "ref": null,
            "rendered": "Lorem",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "scroll scrollHasTitle",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ModalComponent should render 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalComponent
    isOpen={true}
    location={null}
    onRequestClose={[Function]}
    title={null}
    type="lorem"
  >
    Lorem
  </ModalComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ariaHideApp": true,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        null,
        <div
          className="scroll"
        >
          Lorem
        </div>,
      ],
      "className": "contentLorem content",
      "closeTimeoutMS": 0,
      "contentLabel": "Modal",
      "isOpen": true,
      "onRequestClose": [Function],
      "overlayClassName": "overlay",
      "parentSelector": [Function],
      "portalClassName": "root",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "scroll",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": true,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          null,
          <div
            className="scroll"
          >
            Lorem
          </div>,
        ],
        "className": "contentLorem content",
        "closeTimeoutMS": 0,
        "contentLabel": "Modal",
        "isOpen": true,
        "onRequestClose": [Function],
        "overlayClassName": "overlay",
        "parentSelector": [Function],
        "portalClassName": "root",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "scroll",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
